kernel/data-pipeline/dialect/mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ingest/client/PasswordEncryption.java

```diff
@@ -107,7 +107,7 @@ private static RSAPublicKey parseRSAPublicKey(final String key) throws GeneralSe
     
     private static byte[] formatKey(final String key) {
         int start = key.indexOf("\n") + 1;
-        int end = key.lastIndexOf("\n");
+        int end = key.endsWith("\n") ? key.substring(0, key.length() - 2).lastIndexOf("\n") : key.lastIndexOf("\n");
         return key.substring(start, end).replace("\n", "").getBytes();
```

Comment:
1, When it needs RSA?

2, Could we just use `key.trim()` to replace `key.substring(start, end)`?

Reply:
the `publicKeyRequested` property determines whether rsa is required

```
    private void handleCachingSha2Auth(final ChannelHandlerContext ctx, final MySQLAuthMoreDataPacket authMoreData) {
        // how caching_sha2_password works: https://dev.mysql.com/doc/dev/mysql-server/8.0.11/page_caching_sha2_authentication_exchanges.html#sect_caching_sha2_info
        if (!publicKeyRequested) {
            if (PERFORM_FULL_AUTHENTICATION == authMoreData.getPluginData()[0]) {
                publicKeyRequested = true;
                ctx.channel().writeAndFlush(new MySQLAuthSwitchResponsePacket(authMoreData.getSequenceId() + 1, new byte[]{REQUEST_PUBLIC_KEY}));
            }
        } else {
            ctx.channel().writeAndFlush(new MySQLAuthSwitchResponsePacket(authMoreData.getSequenceId() + 1,
                    PasswordEncryption.encryptWithRSAPublicKey(password, seed,
                            serverInfo.getServerVersion().greaterThanOrEqualTo(8, 0, 5) ? "RSA/ECB/OAEPWithSHA-1AndMGF1Padding" : "RSA/ECB/PKCS1Padding",
                            new String(authMoreData.getPluginData()))));
        }
    }
```

2. seem i can use `trim()` before passing in the key parameter

Reply:
1, OK. Could we add unit test for `parseRSAPublicKey`? Since it doesn't occur every time in E2E test